### ISSUE TITLE
select editable: defaultValue is not available until document has been reloaded

### ISSUE THREAD

**Issue by cadeyrn** [pimcore/pimcore#6458]
### Steps to reproduce / Code example:

Let's assume an areabrick with the following code:

```
{% set fooSelect = pimcore_select('foo', {
	store: [
		['option 1', 'option 1'],
		['option 2', 'option 2']
	],
	defaultValue: 'option 1',
	reload: false
}) %}

{{ fooSelect|raw }}

{{ dump(fooSelect.getData()) }}
```

### Actual behavior

After inserting this areabrick into a document you get the following output:

`null`

After reloading the document (you can use the green reload icon in the sidebar; you don't have to save or leave and re-open the document) you get the following outout:

`"option 1"`

### Expected behavior

The output should be `"option 1"` directly after inserting the areabrick into the document. If I change the code to the following (not using the defaultValue parameter but setDataFromResource instead) it works as expected:

```
{% set fooSelect = pimcore_select('foo', {
	store: [
		['option 1', 'option 1'],
		['option 2', 'option 2']
	],
	reload: false
}) %}
{% do fooSelect.isEmpty() ? fooSelect.setDataFromResource('option 1') %}

{{ fooSelect|raw }}

{{ dump(fooSelect.getData()) }}
```

Pimcore version: 6.6.4

**Comment by cadeyrn** [comment]
The bug still exists in Pimcore 6.7.1. Can you please take a look? I would like to know if I have migrate my projects back to setDataFromResource() because it's the source of many problems with the content filling.

Thanks!

Screenshots for reference:

After inserting:

<img width="871" alt="Bildschirmfoto 2020-09-04 um 11 44 10" src="https://user-images.githubusercontent.com/1100614/92225810-0f7b2f80-eea4-11ea-9eb6-338d2062379b.png">

After reloading the document:

<img width="872" alt="Bildschirmfoto 2020-09-04 um 11 44 24" src="https://user-images.githubusercontent.com/1100614/92225827-1609a700-eea4-11ea-9dcb-93274e72f0ea.png">

As you can see the difference is the value of the "text" property. As explained in the original report it only happens with the defaultValue option, the same problem does not exist with the setDataFromResource() method.

**Comment by brusch** [comment]
Please be patient, the issue wasn't closed yet, so yes, why should it be fixed then in 6.7.1? 
If it is urgent for you we're happy to get, review and merge a PR 😉

**Comment by cadeyrn** [comment]
Sorry, I did not want to give the impression of being impatient. I only know from other software projects that sometimes issues get lost in the masses of other issues. And in these projects it's not unusal to check open issues a few months later and give a status update whether the issue still exists or not.

I don't need to have a fix as soon as possible since there is a workaround. It was more meant as "did you noticed it and can you verify that there is a bug or do I have wrong expectations?".

Sorry again, I was not clear enough. 😊 The same applies to my other issue as well, of course. I only write it once to avoid notification spam for others. 😉

**Comment by dvesh3** [comment]
@cadeyrn default value option sets the initial value for the select editable. So it needs to be saved to be fetched from getData() (on reloading a document, saveToSession is called which sets the data temporarily).

If you want to use default value as a fallback, please use `setDataFromResource`.

**Comment by brusch** [comment]
So can be closed?

**Comment by cadeyrn** [comment]
Does it really make sense to differentiate between default and fallback? I mean… the point of a default value is to be the default, no? That's why it's called default.

Especially since the use of setDataFromResource() was replaced with the defaultValue option in the documentation it implies that it should work the same way as before.

**Comment by brusch** [comment]
I guess this just needs better documentation.

**Comment by brusch** [comment]
Fixed by #7477
